modules/ignite-cli-2/src/main/java/org/apache/ignite/cli/core/repl/executor/ReplExecutor.java

```diff
@@ -113,6 +117,8 @@ public SystemRegistryImpl createRegistry() {
     }
 
     public PicocliCommands createPicocliCommands(Class<?> commandClass) {
-        return new PicocliCommands(new CommandLine(commandClass, factory));
+        CommandLine cmd = new CommandLine(commandClass, factory);
+        cmd.setDefaultValueProvider(new CommandLine.PropertiesDefaultProvider(config.getProperties()));
```

Comment:
Will this work if I run REPL with one config, say "ignite.cluster-url=localhost", than I change this config entry to "ignite.cluster-url=myremote" and execute in the same REPL "ignite confi read". What parameter will be taken in this case?

Reply:
Yes, why wouldn't it work? The config is a singleton so changes will be visible.